waku/waku_archive/driver/postgres_driver/postgres_driver.nim

```diff
@@ -571,91 +558,147 @@ proc getMessagesPreparedStmt(
-    let storeTime = $cursor.get().storeTime
+      if ascOrder: SelectNoCursorNoDataAscStmtDef else: SelectNoCursorNoDataDescStmtDef
 
     (
       await s.readConnPool.runStmt(
         stmtName,
         stmtDef,
-        @[contentTopic, pubsubTopic, storeTime, digest, startTimeStr, endTimeStr, limit],
+        @[contentTopic, hashes, pubsubTopic, startTimeStr, endTimeStr, limit],
         @[
           int32(contentTopic.len),
+          int32(hashes.len),
           int32(pubsubTopic.len),
-          int32(storeTime.len),
-          int32(digest.len),
           int32(startTimeStr.len),
           int32(endTimeStr.len),
           int32(limit.len),
         ],
-        @[int32(0), int32(0), int32(0), int32(0), int32(0), int32(0), int32(0)],
+        @[int32(0), int32(0), int32(0), int32(0), int32(0), int32(0)],
         rowCallback,
       )
     ).isOkOr:
-      return err("failed to run query with cursor: " & $error)
-  else:
-    var stmtName =
-      if ascOrder: SelectNoCursorV2AscStmtName else: SelectNoCursorV2DescStmtName
-    var stmtDef =
-      if ascOrder: SelectNoCursorV2AscStmtDef else: SelectNoCursorV2DescStmtDef
+      return err(stmtName & ": " & $error)
 
-    (
-      await s.readConnPool.runStmt(
-        stmtName,
-        stmtDef,
-        @[contentTopic, pubsubTopic, startTimeStr, endTimeStr, limit],
-        @[
-          int32(contentTopic.len),
-          int32(pubsubTopic.len),
-          int32(startTimeStr.len),
-          int32(endTimeStr.len),
-          int32(limit.len),
-        ],
-        @[int32(0), int32(0), int32(0), int32(0), int32(0)],
-        rowCallback,
-      )
-    ).isOkOr:
-      return err("failed to run query without cursor: " & $error)
+    return ok(rows)
+
+  let hashHex = toHex(cursor.get())
+
+  let timeCursor = ?await s.getTimeCursor(hashHex)
+
+  if timeCursor.isNone():
+    return err("cursor not found")
+
+  let timeString = $timeCursor.get()
+
+  var stmtName =
+    if ascOrder:
+      SelectWithCursorNoDataAscStmtName
+    else:
+      SelectWithCursorNoDataDescStmtName
+  var stmtDef =
+    if ascOrder: SelectWithCursorNoDataAscStmtDef else: SelectWithCursorNoDataDescStmtDef
+
+  (
+    await s.readConnPool.runStmt(
+      stmtName,
+      stmtDef,
+      @[
+        contentTopic, hashes, pubsubTopic, hashHex, timeString, startTimeStr,
+        endTimeStr, limit,
+      ],
+      @[
+        int32(contentTopic.len),
+        int32(hashes.len),
+        int32(pubsubTopic.len),
+        int32(timeString.len),
+        int32(hashHex.len),
+        int32(startTimeStr.len),
+        int32(endTimeStr.len),
+        int32(limit.len),
+      ],
+      @[int32(0), int32(0), int32(0), int32(0), int32(0), int32(0), int32(0), int32(0)],
+      rowCallback,
+    )
+  ).isOkOr:
+    return err(stmtName & ": " & $error)
 
   return ok(rows)
 
 method getMessages*(
     s: PostgresDriver,
-    includeData = false,
-    contentTopicSeq = newSeq[ContentTopic](0),
+    includeData = true,
```

Comment:
any specific reason why we change the default?

Reply:
Not really, I feel like this case would happen more often. WDYT?

Reply:
Sounds good :))